dev/src/reference.ts

```diff
@@ -867,15 +867,76 @@ interface QueryCursor {
   values: unknown[];
 }
 
-/** Internal options to customize the Query class. */
-interface QueryOptions {
-  startAt?: QueryCursor;
-  startAfter?: QueryCursor;
-  endAt?: QueryCursor;
-  endBefore?: QueryCursor;
-  limit?: number;
-  offset?: number;
-  projection?: api.StructuredQuery.IProjection;
+/**
+ * Internal class representing custom Query options.
+ *
+ * These options are immutable. Modified options can be created using `with()`.
+ * @private
+ */
+export class QueryOptions {
+  constructor(
+      readonly parentPath: ResourcePath, readonly collectionId: string,
+      readonly allDescendants: boolean, readonly fieldFilters: FieldFilter[],
+      readonly fieldOrders: FieldOrder[], readonly startAt?: QueryCursor,
+      readonly endAt?: QueryCursor, readonly limit?: number,
```

Comment:
It looks like `startAfter` and `endBefore` were omitted, what is the reason? Looks like they were unused previously?

Reply:
This is actually meant to mimic the proto, where `startAfter` and `endBefore` are just different settings in the QueryCurso (`startAt` sets before to `true`, whereas `startAfter` sets it to `false`).